python/lsst/summit/utils/guiders/seeing.py

```diff
@@ -0,0 +1,517 @@
+                ["R00_SG0", "R00_SG1"],
+                ["R04_SG0", "R04_SG1"],
+                ["R40_SG0", "R40_SG1"],
+                ["R44_SG0", "R44_SG1"],
+            ],
+            "cross": [
+                ["R00_SG0", "R40_SG1"],
+                ["R04_SG1", "R44_SG0"],
+                ["R00_SG1", "R04_SG0"],
+                ["R40_SG0", "R44_SG1"],
+            ],
+            "diagonal": [
+                ["R00_SG0", "R44_SG1"],
+                ["R00_SG1", "R44_SG0"],
+                ["R40_SG0", "R04_SG1"],
+                ["R40_SG1", "R04_SG0"],
+            ],
+        }
+        return presets[kind]
+
+    def measureCorrPairs(
+        self, kind: str
+    ) -> tuple[NDArray[np.float64], NDArray[np.float64], NDArray[np.float64]]:
+        """
+        Compute pairwise correlations, covariances, and differential
+        variances between centroid motions of detector pairs.
+
+        The detector pairs are chosen by a preset specified by ``kind``.
+        For each pair of detectors, star centroids are rotated into the
+        local (perpendicular, parallel) frame defined by the vector that
+        connects their mean positions. The method then computes:
+        - Pearson correlation coefficients,
+        - covariance (scaled to centipix²), and
+        - variance of the centroid differences (scaled to centipix²),
+        separately for the perpendicular (x) and parallel (y) components.
+
+        Parameters
+        ----------
+        kind : `str`
+            Preset name passed to :meth:`makePairs`.
+
+        Returns
+        -------
+        corrs : `ndarray`
+            Shape (P, 2). Correlation coefficients per pair:
+            columns ``[corrPerp, corrPar]``.
+        covs : `ndarray`
+            Shape (P, 2). Covariances per pair (scaled by ``100^2``):
+            columns ``[covPerp, covPar]``.
+        diffVars : `ndarray`
+            Shape (P, 2). Variances of centroid differences per pair
+            (scaled by ``100^2``): columns ``[diffVarPerp, diffVarPar]``.
+        """
+        pairs = self.makePairs(kind)
+
+        corrs: list[list[float]] = []
+        covs: list[list[float]] = []
+        diffVars: list[list[float]] = []
+
+        for det1, det2 in pairs:
+            df1 = self.starList.get(det1)
+            df2 = self.starList.get(det2)
+            if df1 is None or df2 is None:
+                corrs.append([np.nan, np.nan])
+                covs.append([np.nan, np.nan])
+                diffVars.append([np.nan, np.nan])
+                continue
+
+            i1, i2 = _alignTimes(df1, df2)
+            if i1 is None or i2 is None:
+                corrs.append([np.nan, np.nan])
+                covs.append([np.nan, np.nan])
+                diffVars.append([np.nan, np.nan])
+                continue
+
+            c1, c2 = _centers(df1, df2)
+            cent1 = _correctedXy(df1)
+            cent2 = _correctedXy(df2)
+            rot1, rot2 = rotateCentroids(cent1, cent2, c1, c2)
```

Comment:
Please use longer, more descriptive variable names so that code further down is easier to read.